ktor-server/ktor-server-cio/jvmAndNix/src/io/ktor/server/cio/CIOApplicationEngine.kt

```diff
@@ -113,6 +117,46 @@ public class CIOApplicationEngine(
         }
     }
 
+    private suspend fun addHandlerForExpectedHeader(output: ByteWriteChannel, call: CIOApplicationCall) {
+        val continueResponse = "HTTP/1.1 100 Continue\r\n"
+        val expectHeaderValue = "100-continue"
+
+        val expectHeader = call.request.headers[HttpHeaders.Expect]?.lowercase()
+        var closeConnection = true
+
+        val expectedHeaderPhase = PipelinePhase("ExpectedHeaderPhase")
+        call.request.pipeline.insertPhaseBefore(ApplicationReceivePipeline.Before, expectedHeaderPhase)
+        call.request.pipeline.intercept(expectedHeaderPhase) {
+            val request = call.request
+            val version = HttpProtocolVersion.parse(request.httpVersion)
+            val alreadyRead = request.receiveChannel().totalBytesRead > 0
```

Comment:
Isn't `receiveChannel` calls `receive()` that triggers `ApplicationReceivePipeline`?

Reply:
Also, how body can be read before this pipeline is executed?

Reply:
Thanks for the comments. I wanted to check if a server has already received some or all of the message body for the request. Delete it now because it's okay to send 100-continue response in such case and I don't know a better solution now